datasync/downloader/downloader.go

```diff
@@ -1113,6 +1113,7 @@ func (d *Downloader) fetchParts(errCancel error, deliveryCh chan dataPack, deliv
 
 			// numTotalPeers means the number of peers satisfying the protocol requirement.
 			idlePeers, numTotalPeers := idlePeers()
+			pendCount := pending()
```

Comment:
Could you explain this statement meaning? i think this statement is not necessary because it is not used until next assignment.

Reply:
There is no next assignment if there is no idlePeers. Does this make sense for your question?

Reply:
Hmm.. what would be changed by this PR? Why does this change "prevent an increase of pending count while escaping on the short cut way"?

Reply:
oh I was wrong.. Thank you for the answer! @jeongkyun-oh

Reply:
@ehnuje It calls the `pending` function on the line 1162. The pending count may be increased after the for loop and it returns an error.